libraries/AP_Mount/SoloGimbal_Parameters.cpp

```diff
@@ -128,22 +131,28 @@ void SoloGimbal_Parameters::update()
     if(!received_all()){
         if (tnow_ms-_last_request_ms > _retry_period) {
             _last_request_ms = tnow_ms;
-            mavlink_msg_param_request_list_send(_chan, 0, MAV_COMP_ID_GIMBAL);
-
-            for(uint8_t i=0; i<MAVLINK_GIMBAL_NUM_TRACKED_PARAMS; i++) {
-                if (!_params[i].seen) {
-                    _params[i].fetch_attempts++;
+            
+            if (HAVE_PAYLOAD_SPACE(_chan, PARAM_REQUEST_LIST)) { 
+                mavlink_msg_param_request_list_send(_chan, 0, MAV_COMP_ID_GIMBAL);
+                
+                for(uint8_t i=0; i<MAVLINK_GIMBAL_NUM_TRACKED_PARAMS; i++) {
+                    if (!_params[i].seen) {
+                        _params[i].fetch_attempts++;
+                    }
                 }
             }
         }
     }
 
     // retry param_set
     for(uint8_t i=0; i<MAVLINK_GIMBAL_NUM_TRACKED_PARAMS; i++) {
-        if (_params[i].state == GMB_PARAMSTATE_ATTEMPTING_TO_SET && tnow_ms - _last_request_ms > _retry_period) {
-            mavlink_msg_param_set_send(_chan, 0, MAV_COMP_ID_GIMBAL, get_param_name((gmb_param_t)i), _params[i].value, MAV_PARAM_TYPE_REAL32);
-            if (!_params[i].seen) {
-                _params[i].fetch_attempts++;
+        if ((_params[i].state == GMB_PARAMSTATE_ATTEMPTING_TO_SET) && (tnow_ms - _last_set_ms > _retry_period)) {
+            _last_set_ms = AP_HAL::millis();
+            if (HAVE_PAYLOAD_SPACE(_chan, PARAM_SET)) {
```

Comment:
This is never going to become true once it becomes false.  Just break out of the loop if you run out of space.